### PR TITLE
Add simple rnn python example

### PR DESCRIPTION
**Pull request by dding3**
## What changes were proposed in this pull request?
Add simple rnn python example

## How was this patch tested?
manual test

### CODE REVIEW

**Comment by dding3** [comment]
Jekins pass. http://10.239.47.211:18888/job/BigDL-PR-Validation/1163/

File: spark/dl/src/main/scala/com/intel/analytics/bigdl/python/api/PythonBigDL.scala [hunk 1]
```diff
@@ -1510,9 +1519,11 @@ class PythonBigDL[T: ClassTag](implicit ev: TensorNumeric[T]) extends Serializab
                     batchSize: Int,
                     trigger: Trigger,
                     valRdd: JavaRDD[Sample],
-                    vMethods: JList[String]): Unit = {
+                    vMethods: JList[String],
+                    criterion: String = null,
```
Comment (zhichao-li) [inline spark/dl/src/main/scala/com/intel/analytics/bigdl/python/api/PythonBigDL.scala]: `criterion` and `embeddedCri` added here for the `loss` validation method only. We might need to convert `vMethods` from `List[String]` to `List[ValidationMethod]` instead of hacking here. You can take a look at the `SGD(JavaValue)` for how to warp those.

File: pyspark/bigdl/models/rnn/README.md [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+      --executor-memory 60g \
+       --py-files ${PYTHON_API_PATH} \
+       --properties-file ${BigDL_HOME}/dist/conf/spark-bigdl.conf \
+       --jars ${BigDL_JAR_PATH} \
+       --archives venv.zip,tokenizers.zip#tokenizers \
+       --conf spark.driver.extraClassPath=${BigDL_JAR_PATH} \
+       --conf spark.executor.extraClassPath=bigdl-version-jar-with-dependencies.jar \
+       --conf spark.yarn.appMasterEnv.NLTK_DATA=./ \
+       --num-executors 1 \
+       ${BigDL_HOME}/pyspark/dl/models/rnn/rnnexample.py --folder hdfs://almaren-node-001:9000/rnn/ --batchSize 12
```
Comment (zhichao-li) [inline pyspark/bigdl/models/rnn/README.md]: add a command for local integration test here ($BigDL/pyspark/test/local_integration/commands)?  we might enable it on jenkins pretty soon.

File: pyspark/bigdl/models/rnn/README.md [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+      3998,3171,3958,2390,3832,3202,3855,3983,3883,3999
+      3998,3667,3999
+      3998,3151,3883,3999
+      3998,3875,3690,3999
+```
+
+## Train the Model
+Example command:
+```bash
+./dist/bin/bigdl.sh -- \
```
Comment (zhichao-li) [inline pyspark/bigdl/models/rnn/README.md]: remove bigdl.sh as it's no need anymore.

File: pyspark/bigdl/models/rnn/README.md [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+The implementation of RNNs in this code is referred to in the [Keras Recurrent](https://keras.io/layers/recurrent/) documentation.
+
+
+## Get the BigDL files
+
+Please build BigDL referring to [Build Page](https://github.com/intel-analytics/BigDL/wiki/Build-Page).
+
+
+## Prepare the Input Data
+You can download the Tiny Shakespeare Texts corpus from [here](https://raw.githubusercontent.com/karpathy/char-rnn/master/data/tinyshakespeare/input.txt).
```
Comment (zhichao-li) [inline pyspark/bigdl/models/rnn/README.md]: Would you like to wrap the data loading same as `$BigDL/pyspark/bigdl/dataset/mnist.py` ?

File: pyspark/bigdl/models/rnn/README.md [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+
+After downloading the text, separate it into train.txt and val.txt. In our case, we just select 80 percentage of the input to be train and remaining 20 percentage to be val.
+Please upload the directory will contains train.txt and val.txt to hdfs. The program will later read in the original text file from hdfs.
+```shell
+export LANG=en_US.UTF-8
+head -n 8000 input.txt > val.txt
```
Comment (zhichao-li) [inline pyspark/bigdl/models/rnn/README.md]: Split train and val via RDD? ``` train_rdd, val_rdd = sample_rdd.randomSplit(
        [training_split, 1-training_split])```
Not a must though.

File: pyspark/bigdl/models/rnn/rnnexample.py [hunk 6]
```diff
@@ -0,0 +1,171 @@
...
+    pad_sent = train_sentences_rdd.flatMap(lambda x: x). \
+        map(lambda sent: sentence.sentences_bipadding(sent))
+    tokens = pad_sent.map(lambda pad: sentence.sentence_tokenizer(pad))
+    max_len = tokens.map(lambda x: len(x)).max()
+    print("max length %s" % max_len)
+
+    words = tokens.flatMap(lambda x: x)
+    print("%s words and %s sentences processed in train data" % (words.count(), tokens.count()))
+
+    val_sentences_rdd = sc.textFile(val_file) \
```
Comment (zhichao-li) [inline pyspark/bigdl/models/rnn/rnnexample.py]: should we extract this as a method? seems like there are some duplicate code between train&val dataset

File: spark/dl/src/main/scala/com/intel/analytics/bigdl/python/api/PythonBigDL.scala (hunk 1 above)
Comment (zhichao-li) [inline spark/dl/src/main/scala/com/intel/analytics/bigdl/python/api/PythonBigDL.scala]: cc @dding3  https://github.com/intel-analytics/BigDL/pull/1149 has implemented it. Please update accordingly.

File: pyspark/bigdl/models/rnn/README.md (hunk 4 above)
Comment (dding3) [inline pyspark/bigdl/models/rnn/README.md]: Updated

**Comment by dding3** [comment]
PR validation pass. http://172.168.2.101:8080/job/BigDL-PR-Validation/1231/

**Comment by zhichao-li** [comment]
squash commits?
